src/main/java/seedu/placebook/logic/commands/EditAppCommand.java

```diff
@@ -69,10 +72,20 @@ public CommandResult execute(Model model) throws CommandException {
         Appointment appointmentToEdit = lastShownList.get(index.getZeroBased());
         Appointment editedAppointment = createEditedAppointment(appointmentToEdit, editAppDescriptor);
 
+        // If End is later than the Start
+        if (editedAppointment.getEnd().compareTo(editedAppointment.getStart()) < 0) {
+            throw new EndTimeBeforeStartTimeException(MESSAGE_END_BEFORE_START
+                    + '\n' + editedAppointment.getEndDateTimeString()
+                    + "\nis before\n" + editedAppointment.getStartDateTimeString());
+        }
+
+        // If clashing appointment list is not empty
         if (!model.getClashingAppointments(editedAppointment).isEmpty()) {
-            if (!model.getClashingAppointments(editedAppointment).contains(appointmentToEdit)) {
+
+            // If the only clashing appointment is itself, skip. the timing may get edited anyway
+            if (model.getClashingAppointments(editedAppointment).size() != 1) {
```

Comment:
I think in this case, maybe can overload the getClashingAppointments method, to take two appointments, one for the appointment to check, one for the appointment being edited, so that you can skip the appointment being edited in the getClashingAppointments method instead.

Reply:
ok i solved it by first creating the list of clashing appointments then removing the original appointment to edit from that list. then for the clashingAppointmentsString i just created the string in the method itself so that don't need to change so much of other code, although not very OOP